vscode-trace-extension/src/utils/trace-extension-webview-manager.ts

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import * as vscode from 'vscode';
+
+/**
+ * Manages webview panels and webviews creation events
+ */
+export class TraceExtensionWebviewManager {
+    private webviews: vscode.WebviewView[] = [];
+    private webviewCreated: vscode.EventEmitter<vscode.WebviewView> = new vscode.EventEmitter();
+    private webviewPanelCreated: vscode.EventEmitter<vscode.WebviewPanel> = new vscode.EventEmitter();
+    private isManagerDisposed = false;
+
+    getAllActiveWebviews(): vscode.WebviewView[] {
+        if (!this.isDisposed()) {
+            return this.webviews;
+        }
+        return [];
+    }
+
+    fireWebviewCreated(_webview: vscode.WebviewView): void {
+        if (!this.isDisposed()) {
+            this.webviewCreated.fire(_webview);
+            this.addWebview(_webview);
```

Comment:
It might be better to add the webview before firing the event, in case adopters have logic that expects the return from `getAllActiveWebviews` to include the just-added webview when they run their handler.